txpipe/diagnostics.py

```diff
@@ -678,6 +672,146 @@ def plot_mag_histograms(self):
             fig.close()
 
 
+class TXLensDiagnosticPlots(PipelineStage):
+    """
+    """
+    name='TXLensDiagnosticPlots'
+    # This tells ceci to launch under dask:
+    dask_parallel = True
+
+    inputs = [
+        ('photometry_catalog', HDFFile),
+        ('lens_tomography_catalog', TomographyCatalog),
+    ]
+
+    outputs = [
+        ('lens_snr_hist', PNGFile),
+        ('lens_mag_hist', PNGFile),
+
+    ]
+
+    config_options = {
+        'chunk_rows': 50000,
+        'delta_gamma': 0.02,
+        'mag_min': 18,
+        'mag_max': 28,
+        'snr_min': 5,
+        'snr_max': 200,
+        'bands': 'ugrizy',
+    }
+
+    def run(self):
+        # PSF tests
+        import matplotlib
+        matplotlib.use('agg')
+
+        files, data, nbin = self.load_data()
+        self.plot_snr_histograms(data, nbin)
+        self.plot_mag_histograms(data, nbin)
+
+        # These need to stay open until the end
+        for f in files:
+            f.close()
+
+
+    def load_data(self):
+        import dask.array as da
```

Comment:
Is there going to be an issue that this function returns dask arrays to a run that doesn't have dask loaded

Reply:
It should be okay - serial-mode dask is always loaded - in that context it basically works as just numpy. The parallel dask isn't always loaded. In general it's fine to pass things through other functions that don't have a module loaded locally.